### DISCUSSION TITLE
[Windows] Mouse events in wezterm or alacritty not captured

### DISCUSSION THREAD

**Discussion by axdank** [helix-editor/helix#3845] [Show and tell] [answered]
Why mouse events are not captured in wezterm or alacritty when using helix (Windows) ?.

**Comment by axdank** [comment, marked as answer]
In many cases the mouse is necessary in helix.

For those who have this problem, it is probably because conhost.exe in windows is outdated.
I have experienced the problem with crossterm which did not detect only the mouse in emulators like wezterm and alacritty.
In windows terminal the mouse works correctly, because it makes internal use of OpenConsole.exe, which is based on the same code as conhost.exe, but more updated.

Therefore a quick solution is to clone the windows terminal repository, compile, and look in the bin folder, the OpenConsole.exe binary, then in the system32 folder, change the permissions of conhost.exe, to our user name with full control, rename it to conhost-old.exe, and is its replacement, copy OpenConsole.exe to the system32 folder and modify its name to conhost.exe.

![imagen](https://user-images.githubusercontent.com/55039048/190307306-883ba556-81ec-4f67-8495-38898208def3.png)

In my case this worked wonders, and solved my mouse problem.

[[Conpty] Add support for mouse input](https://github.com/microsoft/terminal/issues/376)

[[Interesting Usage] Replace the Windows default conhost.exe to OpenConsole.exe](https://github.com/microsoft/terminal/issues/1817)

[Mouse not working (Windows + Alacritty) - Helix Issue](https://github.com/helix-editor/helix/issues/2446)

**Comment by sploders101** [comment]
Awesome solution, thank you! For anyone who doesn't want to build from source, the `OpenConsole.exe` file comes with a binary installation of Windows Terminal, as well. For me, it was located under:

`C:\Program Files\WindowsApps\Microsoft.WindowsTerminal_1.19.10821.0_x64__8wekyb3d8bbwe\OpenConsole.exe`
